### PR TITLE
Add message embed capability

### PR DESCRIPTION
**Pull request by cfpinto**
Added ability to convert URLs inside messages to embed objects. Messages like: 
 **"Watch this video https://www.youtube.com/watch?v=jNQXAC9IVRw"** will be converted into youtube embed markup, where the message is rendered using:
```blade 
{{$message->toHtmlString()}}
```

### CODE REVIEW

File: config/talk.php [hunk 1]
```diff
@@ -19,4 +19,9 @@
             ]
         ],
     ],
+    'oembed' => [
+        'enabled' => false,
+        'url' => '',
```
[the remark this answers is not in the document]
Comment (cfpinto) [reply config/talk.php]: To use with the custom provider. AKA private oembed, useful to provide embed images and file downloads

File: src/Messages/Message.php [hunk 2]
```diff
@@ -2,9 +2,12 @@
 
 namespace Nahid\Talk\Messages;
 
+use Illuminate\Contracts\Support\Htmlable;
```
[the remark this answers is not in the document]
Comment (cfpinto) [reply src/Messages/Message.php]: yes, it's the return type of the toHtmlString method as described in the docblock.

File: README.md [hunk 3]
```diff
@@ -534,6 +542,57 @@ return [
 ];
 ```
 
+## Oembed support
+
+Tall also supports embed urls simply use `$message->toHtlmString()` in you views to render an embed link
```
[the remark this answers is not in the document]
Comment (cfpinto) [reply README.md]: You are right. Will change and push again

**Comment by cfpinto** [comment]
All recommendations answered. All tests passing

File: config/talk.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (nahid) [reply config/talk.php]: So, what will be when we want to develop multiple provider?

**Comment by nahid** [comment]
@cfpinto  In config `oembed` section should contains others options like 'width`, `height`. So it will be more configurable.

**Comment by cfpinto** [comment]
The oembed specifications already cover that level of customization. If having styling problems css should be able to handle it. The custom oembed driver does not support offline embed so width and height not necessary I believe

**Pull request merged**